src/Microsoft.Extensions.DependencyInjection/ServiceLookup/ClosedIEnumerableService.cs

```diff
@@ -3,19 +3,18 @@
 
 using System;
 using System.Collections.Generic;
-using System.Linq;
 
 namespace Microsoft.Extensions.DependencyInjection.ServiceLookup
 {
     internal class ClosedIEnumerableService : IService
     {
-        private readonly Type _itemType;
         private readonly ServiceEntry _serviceEntry;
 
         public ClosedIEnumerableService(Type itemType, ServiceEntry entry)
         {
-            _itemType = itemType;
+            ServiceType = itemType;
             _serviceEntry = entry;
+            ImplementationType = itemType.MakeArrayType();
```

Comment:
Makes sense. It just seems likely that the ImplementationType could become out of sync with the real implementation type if we optimize this later using a different IEnumerable implementation.

This is especially true considering the call to create the array is in CallSiteExpressionBuilder/RuntimeResolver and not in this class. I guess this lack of locality is a downside of the visitor pattern in general.

No matter what IEnumerable implementation we use, it's unlikely to be disposable, but who knows what else we might use IService.ServiceType for.